plugins/package-managers/node/src/funTest/assets/projects/synthetic/yarn/invalid-package-json-expected-output.yml

```diff
@@ -0,0 +1,217 @@
+---
```

Comment:
commit: When the issue was discovered, was the malformed `package.json` in the analyzed project's repository, or in the repository of a dependency?

Can you give a hint why that `package.json` was previously not installed under `node_modeuls`, but now is?

Reply:
> When the issue was discovered, was the malformed package.json in the analyzed project's repository, or in the repository of a dependency?

As shown in the test it was in the repository of the dependency. I also mentioned this in the commit message.

Reply:
> I've read the commit message, but could not figure out this info and still cannot.

This is the part of the commit message where I linked to the package that has this malformed package.json:

```
This causes an error if a test of a dependency contains a malformed
`package.json` file, like e.g. https://github.com/browserify/resolve/blob/v1.22.2/test/resolver/malformed_package_json/package.json
```

>  Can you tell which dependency has the problematic package.json ?

It is `resolve`: https://github.com/browserify/resolve
Here is the malformed package.json: https://github.com/browserify/resolve/blob/v1.22.2/test/resolver/malformed_package_json/package.json

this is the dependency that is used in the added FunTest:
```
  "dependencies": {
    "resolve": "1.22.2"
  }
```

Reply:
> I've made an implementation of what I had in mind and created a draft here: https://github.com/oss-review-toolkit/ort/pull/10285

Thanks. Yes, your changes seem correct. I added them to this PR with some slight adjustments.

The required changes that I made to `babel-expected-output.yml` also look correct.

Reply:
Am I correct to assume that because the actual fix is now done in the common `Utils.kt`, the fix applies to *all* Node package managers?

Reply:
> The required changes that I made to `babel-expected-output.yml` also look correct.

I recall that my tests passed without these modifications. But I must have missed something.

Reply:
> > The required changes that I made to `babel-expected-output.yml` also look correct.
> 
> I recall that my tests passed without these modifications. The babel fun test also fails. Maybe try without modifications to babel expected output.

I checked out your branch from https://github.com/oss-review-toolkit/ort/pull/10285, and the test for babel failes.

Reply:
> I checked out your branch from #10285, and the test for babel fails as well. So the changes are necessary (and IMO more correct than before)

Strange, then I probably have made a mistake. Anyhow, the CI job will be source of truth anyway.